src/language-js/print/assignment.js

```diff
@@ -224,6 +233,19 @@ function isSimpleCall(node) {
   );
 }
 
+function isObjectPropertyWithShortKey(node, keyDoc, options) {
+  if (
+    (node.type !== "ObjectProperty" && node.type !== "Property") ||
+    !node.key
+  ) {
+    return false;
+  }
+  // TODO: a more lightweight version of cleanDoc is needed that would stop once it's
+  // detected that the doc can't be normalized into a string.
+  keyDoc = cleanDoc(keyDoc);
```

Comment:
Use `printDocToString`?

Reply:
Doesn't sound like an optimization

Reply:
Doesn't, but don't relay on how the doc is build.

Reply:
`printDocToString` always returns a string, but I need to check that the doc consists only of strings. No line breaks, etc. How is that worse than, for example, `willBreak`? Doc structure checks are everywhere in Prettier's code.

Reply:
For example `label`.

Reply:
`label` is for complex things like member chains. We definitely don't want to consider such keys "simple".